### PR TITLE
Feature/37385 image lightbox

### PR DESCRIPTION
**Pull request by fabclj**
This PR implements the possibility to left-click on one image preview and open a lightbox/modal containing the image in full size, responsive. On top right there is the button for close the modal and download the original file.

**How to test:**
Create a inbox conversation with LA and use it to send a message with image.

### CODE REVIEW

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx
Comment (sushmi21) [inline src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx:61]: Would be nice to make this button accessible by 
1. Adding visible focus style
2. Adding meaningful aria-label

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx
Comment (sushmi21) [inline src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx:64]: Would be better to use button element instead of anchor element and style it similar to the download button. Else, use role="button".

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx
Comment (sushmi21) [inline src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx:64]: The buttons still don't receive keyboard focus. We need to add tabIndex=0

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx
Comment (sushmi21) [inline src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx:13]: i think it would be nice to use the primary color from the theme and to style the buttons in the header in the same way as that of the webchat header

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/ImagePreview.tsx
Comment (sushmi21) [inline src/plugins/messenger/MessengerPreview/components/MessengerImage/components/ImagePreview.tsx:41]: We can improve the focus style of these images and make it consistent with the rest of the elements. We can remove the outline and add an inset box shadow that uses the primary weak color from the theme (For example. 'box-shadow: inset 0 0 0px 1px <theme.primaryWeakColor>'. Similar to focusing audio and video elements)

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/ImagePreview.tsx
Comment (sushmi21) [inline src/plugins/messenger/MessengerPreview/components/MessengerImage/components/ImagePreview.tsx:54]: Now that we add tabIndex=0, we must add role="button" and an aria-label to this wrapper. Please do not add the button role and aria-label to FlexImage component, since it uses the image element. We can instead wrap the FlexImage in a div that acts as a button. We need to be sure that adding this div does not break anything.

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx
Comment (sushmi21) [inline src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx:6]: When I open the lightbox and tab through it, only the download and close buttons should receive keyboard focus (That is, focus should be trapped within the lightbox). But all the interactive elements in the webchat are receiving focus, which should not be the case.

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/ImagePreview.tsx
Comment (fabclj) [reply src/plugins/messenger/MessengerPreview/components/MessengerImage/components/ImagePreview.tsx]: right now I just added some opacity on hover and focus. Can be improved later on next commit

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx
Comment (sushmi21) [inline src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx:50]: We can use the primaryStrongColor from the theme

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx
Comment (sushmi21) [inline src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx:24]: We should use primaryContrastColor from the theme, as the text will be hard to read on light backgrounds (White or other light colors might also be the primary color).

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx
Comment (sushmi21) [inline src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx:43]: Same here, we should use primaryContrastColor from these, so that it is visible on white backgrounds. Also same for 'fill' in line 47.

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/ImagePreview.tsx
Comment (sushmi21) [inline src/plugins/messenger/MessengerPreview/components/MessengerImage/components/ImagePreview.tsx:53]: The key down handler should also be moved to the div

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/ImagePreview.tsx
Comment (sushmi21) [inline src/plugins/messenger/MessengerPreview/components/MessengerImage/components/ImagePreview.tsx:29]: I tested the focus style, I think we can leave it with the opacity instead of border or box-shadow. But also please add the same focus style for FlexImage

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/ImagePreview.tsx
Comment (sushmi21) [inline src/plugins/messenger/MessengerPreview/components/MessengerImage/components/ImagePreview.tsx:44]: This won't work. The focus style should be added to the div wrapping this with tabIndex=0

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx
Comment (sushmi21) [inline src/plugins/messenger/MessengerPreview/components/MessengerImage/components/LightBoxHeader.tsx:88]: tabIndex is not needed anymore since we changed the Icons to be a button. Same for the close button

**Review by sushmi21 — APPROVED** [review]
Let us not merge this and wait for approval from Thomas regarding the expansion of images from Say node. Rest looks really good.

File: src/plugins/messenger/MessengerPreview/components/MessengerImage/components/ImagePreview.tsx
Comment (sushmi21) [inline src/plugins/messenger/MessengerPreview/components/MessengerImage/components/ImagePreview.tsx:40]: We need to remove the pointer for FlexImage if the image is not downloadable

**Review by sushmi21 — APPROVED** [review]

**Pull request merged**
